src/components/TokenBalance.tsx

```diff
@@ -79,56 +74,65 @@ const TokenBalance: FC<TokenBalanceProps & TextProps> = ({
   tokenAmount,
   usdBalance,
   tokenSymbol,
-  withUSDBalance = false,
-  withSymbol = false,
   tokenDecimals,
   isLarge,
   tokenFormat,
   precision,
   higherPrecision,
-  withHigherPrecision = false,
+  children,
   ...restProps
 }) => {
   const { active } = useWeb3React()
   const shouldRenderTokenAmount = active
 
   const _tokenAmount = useMemo(() => {
-    return formatTokenAmount(tokenAmount || 0, tokenFormat, tokenDecimals)
+    return formatTokenAmount(
+      tokenAmount || 0,
+      tokenFormat,
+      tokenDecimals,
+      precision
+    )
   }, [tokenAmount, tokenFormat, tokenDecimals])
 
-  const BalanceTag = isLarge ? H3 : H5
-  const SymbolTag = isLarge ? BodyLg : BodySm
-  const usdBalanceColor = useColorModeValue("gray.500", "gray.300")
-
   // TODO: more flexible approach to style wrapper, token balance and USD balance.
   return (
     <Box>
       <HStack alignItems="center">
-        <BalanceTag {...restProps}>
+        <Text
+          fontSize={isLarge ? "4.5xl" : "2xl"}
+          lineHeight={isLarge ? 12 : "lg"}
+          {...restProps}
+        >
           {shouldRenderTokenAmount ? (
-            withHigherPrecision ? (
+            higherPrecision ? (
               <InlineTokenBalance
                 tokenAmount={tokenAmount}
                 tokenFormat={tokenFormat}
                 tokenDecimals={tokenDecimals}
-                precision={precision}
-                withHigherPrecision
+                precision={higherPrecision}
```

Comment:
We should pass `precision` here and add `higherPrecision` too

```suggestion
                precision={precision}
                higherPrecision={higherPrecision}
```

Reply:
Ref commit: b11bf6cb

Reply:
I think you forgot to change `precision={higherPrecision}` to `precision={precision}`. 

Should be:
```
  precision={precision}
  higherPrecision={higherPrecision}
```

Right now it's:
```
  precision={higherPrecision}
  higherPrecision={higherPrecision}
```

Reply:
Ref commit: 33dbe121